src/daemon/config.rs

```diff
@@ -700,68 +703,52 @@ impl Config {
         config
     }
 
-    pub fn get_config_filename() -> String {
-        let matches = App::new(KRILL_SERVER_APP)
-            .version(KRILL_VERSION)
-            .arg(
-                Arg::with_name("config")
-                    .short("c")
-                    .long("config")
-                    .value_name("FILE")
-                    .help("Override the path to the config file (default: './defaults/krill.conf')")
-                    .required(false),
-            )
-            .get_matches();
+    /// Creates the config (at startup).
+    pub fn create(config_file: &str, upgrade_only: bool) -> Result<Self, ConfigError> {
+        let mut config = Self::read_config(config_file)?;
 
-        let config_file = matches.value_of("config").unwrap_or(KRILL_DEFAULT_CONFIG_FILE);
+        if upgrade_only {
+            config.log_type = LogType::Stderr;
+            config.log_level = LevelFilter::Info;
+        }
 
-        config_file.to_string()
-    }
+        config.init_logging()?;
 
-    /// Creates the config (at startup). Panics in case of issues.
-    pub fn create() -> Result<Self, ConfigError> {
-        let config_file = Self::get_config_filename();
+        if upgrade_only {
+            info!("Prepare upgrade using configuration file: {}", config_file,);
+            info!("Processing data from: {}", config.data_dir.to_string_lossy());
+            info!(
+                "Saving prepared data to: {}",
+                config.upgrade_data_dir().to_string_lossy()
+            );
+            warn!("*** Kill this process if you have doubts about available disk space! ***");
```

Comment:
I have no good way to guess.. so that's why it's left as an exercise to the operator... I could also leave out the warning altogether but that also seemed wrong. In most cases this really shouldn't be an issue, and people should monitor disk space.. suggestion?

Reply:
Why is this different than any other time that Krill might encounter errors writing to disk? Why mention it at all?